word-guess-main.rb

```diff
@@ -0,0 +1,86 @@
+
+
+class Game
+
+  def initialize(input_word, word_descriptions)
+    @word = input_word.chars.to_a
+    @hidden_word = @word.join.gsub(/[A-Za-z]/, "_")
+    @bad_guesses = 0
+    @word_descriptions = word_descriptions
+  end
+
+  def guess(letter)
+    if @word.include?(letter) == false
+      @bad_guesses += 1
+      puts "I'm sorry, that letter is not in this word!"
+    else
+      puts "Excellent choice!!"
+      until @word.include?(letter) == false
+        index = @word.index(letter)
+        @word[index] = "*"
+        @hidden_word[index] = letter
+      end
+    end
+  end
+
+  def word_descriptions
+    return "#{@word_descriptions}"
+  end
+
+  def print_guessed_word # prints hidden word
+    puts " Word: #{@hidden_word}"
+  end
+
+  def print_picture
+    count = 5 - @bad_guesses
+    puts "(@)" * count
+    puts "  ,\\,\\,|,/,/,"
+    puts "     _\\|/_"
+    puts "    |_____|"
+    puts"     |   |"
+    puts "     |___|"
+  end
+
+  def is_game_lost
+    if @bad_guesses >= 5
+      return true
+    else
+      return false
+    end
+  end
+
+  def is_game_won
+    if @hidden_word.include?("_")
+      return false
+    else
+      return true
+    end
+  end
+
+
+end
+
+# Main program
+word_array = ["towel", "plant", "apples", "yoga", "banana", "mississippi"]
+word_descriptions = ["A household item", "It's alive!", "You can eat it!", "It's a sport.", "It's a fruit.", "It's a state."]
+rand_num = rand(0..(word_array.length - 1))
+g1 = Game.new(word_array[rand_num], word_descriptions[rand_num])
+
+puts g1.word_descriptions
+
+while true
+  puts "***********"
+  g1.print_guessed_word
+  puts "***********"
```

Comment:
The body of this loop might be a good candidate to turn into a method - something like `take_turn`. The code's already DRY, but it never hurts to put a label on things.